client/containers/Builder/Builder.js

```diff
@@ -22,18 +36,49 @@ class Builder extends Component {
     }
   }
 
+  pedalAddHandler = (id, event) => {
+    const findPedal = this.state.pedals.find((elem, index) => {
+      if (elem.id === id) {
+        return elem
+      }
+    })
+
+    const updatePedalsOnBoard = [...this.state.pedalsOnBoard, findPedal]
+
+    this.setState({ showModal: false, pedalsOnBoard: updatePedalsOnBoard })
+  }
+
+  openModalHandler = () => {
+    this.setState({ showModal: true })
+  }
+
+  closeModalHandler = () => {
+    this.setState({ showModal: false })
+  }
+
   render() {
-    let pedalBoardBuilder = this.state.currentPedalboard ? (
+    const { pedals, showModal, currentPedalboard, pedalsOnBoard } = this.state
+    let pedalBoardBuilder = currentPedalboard ? (
       <Fragment>
-        <BuilderTitle pedalboardName={this.state.currentPedalboard} />
-        <PedalboardBuilderDisplay
-          currentPedalboard={this.state.currentPedalboard}
-        />
+        <BuilderTitle pedalboardName={currentPedalboard} />
+        <PedalboardBuilderDisplay currentPedalboard={currentPedalboard} />
       </Fragment>
     ) : (
       <WarningMessage />
     )
-    return <Fragment>{pedalBoardBuilder}</Fragment>
+    return (
+      <Fragment>
+        <BuilderAddPedalButton showButton={currentPedalboard} showModal={this.openModalHandler} />
+        <BuilderModal
+          closeModalHandler={this.closeModalHandler}
+          showModal={showModal}
+          pedalData={pedals}
+          handleClick={this.pedalAddHandler}
+        />
+        {pedalBoardBuilder}
```

Comment:
Use conditional rendering instead of temporary variables. 

```jsx
{ currentPedalboard &&
  <Fragment>
    <BuilderTitle pedalboardName={currentPedalboard} />
    <PedalboardBuilderDisplay currentPedalboard={currentPedalboard} />
  </Fragment>
}
{ !currentPedalbard &&
  <WarningMessage/>
}
```